client/js/widgets/pile.js

```diff
@@ -130,6 +130,14 @@ class Pile extends Widget {
       await this.parent.dispenseCard(child);
   }
 
+  async onPropertyChange(property, oldValue, newValue) {
+    if(this.children().length && property == 'owner') {
+      for(const c of this.children())
+        await c.set('owner', newValue);
+    }
+    super.onPropertyChange(property, oldValue, newValue);
```

Comment:
This probably needs an `await`? _Might_ actually be the reason for your fragments?

Reply:
I'm not very well versed in the use of async and await. We'll se if this helps (it will be a few hours before I get back to my local server).